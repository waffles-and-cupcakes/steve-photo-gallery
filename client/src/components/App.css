.banner-img {
  position: relative;
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
  height: 75vh;
  cursor: pointer;
}

.view-photos {
  position: absolute;
  bottom: 20px;
  left: 20px;
  padding: 10px 15px;
  border-radius: 3px;
  font-size: .9rem;
}

.modal-open {
  overflow: hidden;
  width: 100%;
  height: 100%;
}