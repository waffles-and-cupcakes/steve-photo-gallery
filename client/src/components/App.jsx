import React from 'react';
import axios from 'axios';
import Modal from './Modal.jsx';
import style from './App.css';

const toggleScrolling = () => {
  const body = document.getElementsByTagName('body')[0];
  body.classList.toggle(style.modalOpen);
};

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
      photos: [
        {
          photoUrl: '',
        },
      ],
    };
    this.toggleModal = this.toggleModal.bind(this);
    this.toggleScrolling = this.toggleScrolling.bind(this);
  }

  componentDidMount() {
    const id = window.location.pathname.split('/')[2];
    axios.get(`/rooms/${id}/photos`)
      .then(res => {
        this.setState({
          photos: res.data,
        });
      });
  }

  toggleModal() {
    this.setState({ isOpen: !this.state.isOpen }, toggleScrolling);
  }

  render() {
    const backgroundStyle = {
      backgroundImage: `url('${this.state.photos[0].photoUrl}')`,
    };
    return (
      <div>
        <div onClick={this.toggleModal} className={style.bannerImg} style={backgroundStyle}>
          <button className={style.viewPhotos}>View Photos</button>
        </div>
        <Modal
          photos={this.state.photos}
          open={this.state.isOpen}
          toggleModal={this.toggleModal} />
      </div>
    );
  }
}

export default App;
