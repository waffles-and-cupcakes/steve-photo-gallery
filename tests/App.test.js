import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import App from './App.jsx';

configure({adapter: new Adapter() });

test('App should render correctly', () => {
  const output = shallow(<App />);
  expect(shallowToJson(output)).toMatchSnapshot();
});

test('toggleModal should toggle isOpen', () => {
  const output = shallow(<App />);
  const btn = output.find('div').get(1);
  btn.props.onClick();
  expect(output.state().isOpen).toBe(true);  
});