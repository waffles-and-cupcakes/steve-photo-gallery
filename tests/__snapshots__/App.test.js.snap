// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url('')",
      }
    }
  >
    <button>
      View Photos
    </button>
  </div>
  <Modal
    open={false}
    photos={
      Array [
        Object {
          "photoUrl": "",
        },
      ]
    }
    toggleModal={[Function]}
  />
</div>
`;
